app/routes.py

```diff
@@ -1,2 +1,293 @@
-from flask import Blueprint
+from flask import Blueprint, request, make_response, jsonify
+from sqlalchemy import asc, desc
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import requests
+import os
+import random
 
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+
+#=====================================================#
+#                    TASK ROUTES                      #
+#=====================================================#
+
+
+@tasks_bp.route("", methods=["POST"])
+def add_new_task():
+    """
+    Create a new Task
+    """
+    request_body = request.get_json()
+
+    try:
+        request_body["title"]
+        request_body["description"]
+        request_body["completed_at"]
+    except:
+        return make_response(jsonify({
+            "details": "Invalid data"
+        }), 400)
+
+    new_task = Task(title=request_body["title"],
+                    description=request_body["description"],
+                    completed_at=request_body["completed_at"])
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return make_response({"task": new_task.to_json()}, 201)
+
+
+@tasks_bp.route("", methods=["GET"])
+def list_all_tasks():
+    """
+    Get all Tasks in asc, desc, or unsorted order
+    """
+    sort_query = request.args.get("sort")
+
+    if sort_query == "asc":
+        tasks = Task.query.order_by(asc("title"))
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(desc("title"))
+    else:
+        tasks = Task.query.all()
+
+    tasks_response = [task.to_json() for task in tasks]
+
+    return jsonify(tasks_response)
+
+
+@tasks_bp.route("/<int:task_id>", methods=["GET"])
+def get_task_by_id(task_id):
+    """
+    Get one Task by id
+    """
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response("Task not found. Less to do then :)", 404)
```

Comment:
You can condense this portion of your code using Flask's `.get_or_404()` method:

```suggestion
    task = Task.query.get_or_404(task_id)

```